src/providers/flagsView.ts

```diff
@@ -493,3 +484,58 @@ export class FlagNode extends vscode.TreeItem {
 		});
 	}
 }
+
+export class FlagParentNode extends vscode.TreeItem {
+	children: FlagNode[] | undefined;
+	contextValue?: string;
+	uri?: string;
+	flagKey?: string;
+	flagParentName?: string;
+	flagVersion: number;
+	enabled?: boolean;
+
+	/**
+	 * @param label will be shown in the Treeview
+	 * @param collapsibleState is initial state collapsible state
+	 * @param children array of FlagNode's building nested view
+	 * @param flagKey reference to which flag key the treeview item is associated with
+	 */
+	constructor(
+		ctx: vscode.ExtensionContext,
+		public readonly label: string,
+		public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+		children?: FlagNode[],
+		flagKey?: string,
+		flagVersion?: number,
+		enabled?: boolean,
+	) {
+		super(label, collapsibleState);
+		this.children = children;
+		this.flagKey = flagKey;
+		this.flagVersion = flagVersion;
+		this.enabled = enabled;
+		this.conditionalIcon(ctx, this.contextValue, this.enabled);
+	}
+
+	private conditionalIcon(ctx: vscode.ExtensionContext, contextValue: string, enabled: boolean) {
+		if (ctx && enabled) {
+			this.setIcon(ctx, 'toggleon');
+		} else if (ctx) {
+			this.setIcon(ctx, 'toggleoff');
+		}
+	}
+
+	private setIcon(ctx: vscode.ExtensionContext, fileName: string): vscode.ThemeIcon {
+		return (this.iconPath = {
+			light: ctx.asAbsolutePath(path.join('resources', 'light', fileName + '.svg')),
+			dark: ctx.asAbsolutePath(path.join('resources', 'dark', fileName + '.svg')),
+		});
+	}
+}
```

Comment:
Are the methods in this new class identical to the ones above? Do we need both?

Reply:
the methods are the same currently as this is step 1. The logic within the methods is getting cleaned up also. I'm also trimming down the number of properties per class and in the future will move additional nodes like `variations` to their own class so it's less of a tangle of procedural code.